src/dataflow/src/server.rs

```diff
@@ -1033,6 +1034,20 @@ where
                     if let Some(ts_history) = self.storage_state.ts_histories.get_mut(&id) {
                         ts_history.set_compaction_frontier(frontier.borrow());
                     }
+
+                    if let Some(compaction_handle) = self.storage_state.compaction_handles.get(&id)
+                    {
+                        let compaction_handle = match compaction_handle.upgrade() {
+                            Some(probe) => probe,
+                            None => {
+                                // Clean up the expired weak handle.
+                                self.storage_state.compaction_handles.remove(&id);
```

Comment:
let's put a TODO that the real way to deal with this is getting a StorageCommand that tells us to stop ingesting this source a la `ComputeCommand::DropIndexes`

Reply:
:+1: